extractor/src/main/java/org/schabi/newpipe/extractor/services/bandcamp/extractors/BandcampCommentsExtractor.java

```diff
@@ -39,18 +51,77 @@ public InfoItemsPage<CommentsInfoItem> getInitialPage()
+        final JsonObject collectorsData = JsonUtils.toJsonObject(
+                document.getElementById("collectors-data").attr("data-blob"));
+        final JsonArray reviews = collectorsData.getArray("reviews");
 
-        for (final Element writing : writings) {
-            collector.commit(new BandcampCommentsInfoItemExtractor(writing, getUrl()));
+        for (final Object review : reviews) {
+            collector.commit(
+                    new BandcampCommentsInfoItemExtractor((JsonObject) review, getUrl()));
         }
 
-        return new InfoItemsPage<>(collector, null);
+        if (!collectorsData.getBoolean("more_reviews_available")) {
+            return new InfoItemsPage<>(collector, null);
+        }
+
+        final String trackId = getTrackId();
+        final String token = getNextPageToken(reviews);
+        return new InfoItemsPage<>(collector, new Page(trackId, token));
     }
 
     @Override
     public InfoItemsPage<CommentsInfoItem> getPage(final Page page)
             throws IOException, ExtractionException {
-        return null;
+
+        final CommentsInfoItemsCollector collector = new CommentsInfoItemsCollector(getServiceId());
+
+        final String trackId = page.getUrl();
+        final String token = page.getId();
+        final JsonObject reviewsData = fetchReviewsData(trackId, token);
+        final JsonArray reviews = reviewsData.getArray("results");
+
+        for (final Object review : reviews) {
+            collector.commit(
+                    new BandcampCommentsInfoItemExtractor((JsonObject) review, getUrl()));
+        }
+
+        if (!reviewsData.getBoolean("more_available")) {
+            return new InfoItemsPage<>(collector, null);
+        }
+
+        final String nextToken = getNextPageToken(reviews);
+        return new InfoItemsPage<>(collector, new Page(trackId, nextToken));
+    }
+
+    private JsonObject fetchReviewsData(final String trackId, final String token)
+            throws ParsingException {
+        try {
+            return JsonUtils.toJsonObject(getDownloader().postWithContentTypeJson(
+                    REVIEWS_API_URL,
+                    Collections.emptyMap(),
+                    JsonWriter.string().object()
+                            .value("tralbum_type", "t")
+                            .value("tralbum_id", trackId)
+                            .value("token", token)
+                            .value("count", 7)
+                            .array("exclude_fan_ids").end()
+                            .end().done().getBytes(StandardCharsets.UTF_8)).responseBody());
+        } catch (final IOException | ReCaptchaException e) {
+            throw new ParsingException("Could not fetch reviews", e);
+        }
+    }
+
+    private String getNextPageToken(final JsonArray reviews) throws ParsingException {
+        return reviews.stream().map(JsonObject.class::cast)
+                .map(o -> o.getString("token"))
+                .reduce((a, b) -> b)
```

Comment:
Please use explicit variable names, `o` should be `review`, but I don't know what is `a` and `b`.

Reply:
Since this line gets the last item, `a` and `b` are probably just supposed to be `first` and `second` like in https://stackoverflow.com/a/21441634/20044811?

Reply:
I think it's fine to have `a` and `b`, but a comment should be added to explain that `.reduce((a, b) -> b)` is just keeping the last element.
```suggestion
                .map(review -> review.getString("token"))
                .reduce((a, b) -> b) // keep only the last element
```